### PR TITLE
Component illumos-gcc shoud use private mpfr, mpc, and gnump.

### PR DESCRIPTION
**Pull request by alarcher**
Testing is WIP.

### CODE REVIEW

File: components/openindiana/illumos-gcc/illumos-gcc.p5m [hunk 1]
```diff
@@ -15,76 +16,76 @@
...
 hardlink \
-    path=opt/gcc/$(COMPONENT_VERSION)/bin/$(GNU_ARCH)-g++ \
+    path=opt/gcc/$(COMPONENT_VERSION)/bin/i386-pc-solaris2.11-g++ \
     target=g++
 hardlink \
-    path=opt/gcc/$(COMPONENT_VERSION)/bin/$(GNU_ARCH)-gcc \
+    path=opt/gcc/$(COMPONENT_VERSION)/bin/i386-pc-solaris2.11-gcc \
     target=gcc
 hardlink \
-    path=opt/gcc/$(COMPONENT_VERSION)/bin/$(GNU_ARCH)-gcc-$(COMPONENT_VERSION) \
```
Comment (pyhalov) [inline components/openindiana/illumos-gcc/illumos-gcc.p5m]: why do you change it?

Comment (alarcher) [reply components/openindiana/illumos-gcc/illumos-gcc.p5m]: I just took the generated manifest actually. I can revert if needed.

Comment (pyhalov) [reply components/openindiana/illumos-gcc/illumos-gcc.p5m]: I'm not sure about this. On one hand, changing manifest when nothing changed seems strange. On the other - it's good when sample-manifest and actually used manifest doesn't differ where they shouldn't. Don't have strong preference here.

File: components/openindiana/illumos-gcc/illumos-gcc.p5m [hunk 2]
```diff
@@ -751,73 +747,116 @@ file path=opt/gcc/$(COMPONENT_VERSION)/lib/$(MACH64)/libstdc++.so.6.0.13 \
...
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/crtbegin.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/crtend.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/gmon.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/libgcc.a
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/libgcc_eh.a
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/$(MACH64)/libgcov.a
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/crtbegin.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/crtend.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/gmon.o
+file path=opt/gcc/$(COMPONENT_VERSION)/lib/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/include-fixed/README
```
Comment (pyhalov) [inline components/openindiana/illumos-gcc/illumos-gcc.p5m]: Don't ship include-fixed!

File: components/openindiana/illumos-gcc/illumos-gcc.p5m [hunk 3]
```diff
@@ -850,15 +885,14 @@ file path=opt/gcc/$(COMPONENT_VERSION)/lib/libstdc++.so.6.0.13 \
...
-file path=opt/gcc/$(COMPONENT_VERSION)/lib/libsupc++.la
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/cc1 mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/cc1obj mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/cc1plus mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/collect2 mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/install-tools/fixinc.sh mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/install-tools/fixincl mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/install-tools/mkheaders mode=0555
-file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/$(GNU_ARCH)/$(COMPONENT_VERSION)/install-tools/mkinstalldirs mode=0555
+file path=opt/gcc/$(COMPONENT_VERSION)/libexec/gcc/i386-pc-solaris2.11/$(COMPONENT_VERSION)/cc1
```
Comment (pyhalov) [inline components/openindiana/illumos-gcc/illumos-gcc.p5m]: You've unintentionally removed mode=0555

**Comment by pyhalov** [comment]
At least, illumos-gate build is fine

File: components/openindiana/illumos-gcc/illumos-gcc.p5m (hunk 1 above)
Comment (alarcher) [reply components/openindiana/illumos-gcc/illumos-gcc.p5m]: I will restore the previous manifest.

**Comment by pyhalov** [comment]
Tested by booting system built with this compiler...

**Pull request merged**